testr/testr_test.go

```diff
@@ -29,14 +29,74 @@ func TestLogger(t *testing.T) {
 	log.V(0).Info("V(0).info")
 	log.V(1).Info("v(1).info")
 	log.Error(fmt.Errorf("error"), "error")
-	log.WithName("testing").Info("with prefix")
+	log.WithName("testing").WithValues("value", "test").Info("with prefix")
+	log.WithName("testing").Error(fmt.Errorf("error"), "with prefix")
 	Helper(log, "hello world")
 
 	log = NewWithOptions(t, Options{
 		LogTimestamp: true,
 		Verbosity:    1,
 	})
 	log.V(1).Info("v(1).info with options")
+
+	underlier, ok := log.GetSink().(Underlier)
+	if !ok {
+		t.Error("couldn't get underlier")
+	}
+	if t != underlier.GetUnderlying() {
+		t.Error("invalid underlier")
+	}
+}
+
+func TestLoggerInterface(t *testing.T) {
+	log := NewWithInterface(t, Options{})
+	log.Info("info")
+	log.V(0).Info("V(0).info")
+	log.V(1).Info("v(1).info")
+	log.Error(fmt.Errorf("error"), "error")
+	log.WithName("testing").WithValues("value", "test").Info("with prefix")
+	log.WithName("testing").Error(fmt.Errorf("error"), "with prefix")
+	Helper(log, "hello world")
+
+	underlier, ok := log.GetSink().(UnderlierInterface)
+	if !ok {
+		t.Fatal("couldn't get underlier")
+	}
+	underlierT, ok := underlier.GetUnderlying().(*testing.T)
+	if !ok {
+		t.Fatal("couldn't get underlying *testing.T")
+	}
+	if t != underlierT {
+		t.Error("invalid underlier")
+	}
+}
+
+func BenchmarkLogger(b *testing.B) {
```

Comment:
do we really need this?  What is it proving?

Reply:
@thockin This is mostly to test that `NewWithInterface` can be called with a `*testing.B` in addition to `*testing.T`.
But it's true that it is not really useful to benchmark this library, so I can remove it if you'd prefer.

Reply:
I see - I think it is important to have test-cases for B and F.

It's probably enough to have something like:


```
b := testing.B{}
log := NewWithInterface(b)
```

and same for F, just to prove it compiles.

Reply:
I added test cases for `*testing.B` and `*testing.F`.
I put the one for `*testing.F` in a separate file since fuzzing is only available since Go 1.18 but the test cases are run on the CI pipeline with Go 1.16+.